compass/ocean/tests/overflow/hydro_vs_nonhydro/initial_state.py

```diff
@@ -0,0 +1,128 @@
+import time
+
+import numpy as np
+import xarray
+from mpas_tools.io import write_netcdf
+from mpas_tools.mesh.conversion import convert, cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+
+from compass.ocean.vertical import init_vertical_coord
+from compass.step import Step
+
+
+class InitialState(Step):
+    """
+    A step for creating a mesh and initial condition for the
+    hydro vs nonhydro test case
+    """
+    def __init__(self, test_case):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+        """
+        super().__init__(test_case=test_case, name='initial_state')
+
+        for file in ['base_mesh.nc', 'culled_mesh.nc', 'culled_graph.info',
+                     'initial_state.nc']:
+            self.add_output_file(file)
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        config = self.config
+        logger = self.logger
+
+        timeStart = time.time()
+
+        section = config['horizontal_grid']
+        nx = section.getint('nx')
+        ny = section.getint('ny')
+        dc = section.getfloat('dc')
+
+        dsMesh = make_planar_hex_mesh(nx=nx, ny=ny, dc=dc, nonperiodic_x=True,
+                                      nonperiodic_y=False)
+        write_netcdf(dsMesh, 'base_mesh.nc')
+
+        dsMesh = cull(dsMesh, logger=logger)
+        dsMesh = convert(dsMesh, graphInfoFileName='culled_graph.info',
+                         logger=logger)
+        write_netcdf(dsMesh, 'culled_mesh.nc')
+
+        section = config['vertical_grid']
+        nVertLevels = section.getint('vert_levels')
+
+        section = config['hydro_vs_nonhydro']
+        xs = section.getfloat('xs')
+        Ls = section.getfloat('Ls')
+        config_eos_linear_Sref = section.getfloat('eos_linear_Sref')
+        config_eos_linear_densityref = section.getfloat(
+            'eos_linear_densityref')
+        rhoz = section.getfloat('rhoz')
+        lower_temperature = section.getfloat('lower_temperature')
+        higher_temperature = section.getfloat('higher_temperature')
+
+        # comment('obtain dimensions and mesh variables')
+        # vertical_coordinate = 'uniform'
+
+        ds = dsMesh.copy()
+        nCells = ds.nCells.size
+        nEdges = ds.nEdges.size
+        nVertices = ds.nVertices.size
+
+        xCell = ds.xCell
+
+        # comment('create and initialize variables')
+        time1 = time.time()
+
+        # bottom depth
+        # ds['bottomDepth'] = maxDepth * xarray.ones_like(xCell)
+        ds['bottomDepth'] = - (- 200.0 + 0.5 * (200.0 - 40.0) *
```

Comment:
```suggestion
        ds['bottomDepth'] = - (- bottom_depth + 0.5 * (bottom_depth - min_depth) *
```
Is this right? If so, add `min_depth` to the cfg file and get both `min_depth` and `bottom_depth` from the cfg file.

Reply:
`bottom_depth` and `shelf_depth` are now read from the cfg file.